sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1069,6 +1069,13 @@ class Analyzer(
       case view: View if !view.child.resolved =>
         ViewResolution
           .resolve(view, options, resolveChild = executeSameContext, checkAnalysis = checkAnalysis)
+      // Resolve V2TableReference nodes in already-resolved views (e.g., temp views storing
+      // analyzed plans). V2TableReference is a placeholder for DSv2 tables that needs to be
+      // resolved to DataSourceV2Relation on each view access.
+      case view: View =>
```

Comment:
This is a safe guard. If the view is already resolved (e.g. df temp view), we don't resolve its body, so it still contains `V2TableReference`. Ideally it's OK as the next analyzer rule execution iteration will replace it, but in case the fixed point happens to be reached, here we eagerly replace it.